java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java

```diff
@@ -485,6 +491,61 @@
                         .thenApply(avoid -> completionFutures.stream().flatMap(c -> c.join().stream()).collect(Collectors.toList()));
                 return (CompletableFuture<Object>) (CompletableFuture<?>) joinedFuture;
             }
+            case Server.JAVA_PROJECT_RESOLVE_PROJECT_PROBLEMS: {
+                CompletableFuture<Object> result = new CompletableFuture<>();
+                List<Object> arguments = params.getArguments();
+                if (!arguments.isEmpty()) {
+                    String fileStr = ((JsonPrimitive) arguments.get(0)).getAsString();
+                    FileObject file;
+                    try {
+                        file = URLMapper.findFileObject(URI.create(fileStr).toURL());
+                    } catch (MalformedURLException ex) {
+                        result.completeExceptionally(ex);
+                        return result;
+                    }
+                    Project project = FileOwnerQuery.getOwner(file);
+                    if (project != null) {
+                        ProjectProblemsProvider ppp = project.getLookup().lookup(ProjectProblemsProvider.class);
+                        if (ppp != null) {
+                            Collection<? extends ProjectProblemsProvider.ProjectProblem> problems = ppp.getProblems();
+                            if (!problems.isEmpty()) {
+                                List<Pair<ProjectProblemsProvider.ProjectProblem, Future<ProjectProblemsProvider.Result>>> resolvers = new LinkedList<>();
+                                for (ProjectProblemsProvider.ProjectProblem problem : ppp.getProblems()) {
+                                    if (problem.isResolvable()) {
+                                        resolvers.add(Pair.of(problem, problem.resolve()));
```

Comment:
is the `.resolve()` called in the LSP message thread ? If so, should be called in a RequestProcessor, since it may eventually trigger a project build.

Reply:
Done.